azure-pipelines.yml

```diff
@@ -397,7 +397,33 @@ jobs:
           # Note: this is going to get the PR branch commit, not the
           # result of the merge (i.e. this is not using the same commit as the
           # other jobs in this build).
-          git fetch origin $(branch_sha)
+          #
+          # We have seen errors getting this commit from GitHub, which we
+          # suspect are transient network errors, so adding simple exponential
+          # backoff to mitigate.
+          BACKOFF=1
+          tell_gary() {
+              curl -XPOST \
+                   -i \
+                   -H 'Content-Type: application/json' \
+                   --data "{\"text\":\"<@UEHSF89AQ> <https://dev.azure.com/digitalasset/daml/_build/results?buildId=$(Build.BuildId)|Build $(Build.BuildId)> for <https://github.com/digital-asset/daml/pull/$(pr.num)|PR $(pr.num)> has failed to fetch its commit $(branch_sha) up to BACKOFF=$BACKOFF.\"}" \
+                   $(Slack.team-daml-ci)
+              fi
+          }
+          trap tell_gary EXIT
+          while !  git fetch origin $(branch_sha); do
```

Comment:
Do we actually need `git fetch` here or only `git checkout`? I would expect that the branch commit should always be in here already. Maybe that would help make this less flaky and it might also explain why we haven’t seen this on steps that use the regular Azure checkout step.

Reply:
We _have_ seen this exact issue on steps that use the regular Azure checkout, as that was the reason for putting in this workaround in the first place.

We _may_ need git fetch because this is a separate job, which means it may run on a separate machine which may not have fetched this commit yet. But now that you mention it maybe we could just check whether we have the commit before trying to fetch it.

Reply:
> We _have_ seen this exact issue on steps that use the regular Azure checkout, as that was the reason for putting in this workaround in the first place.

We have but it seemed like new behavior so maybe whatever change required this in Azure is no longer necessary.

Reply:
True. Hopefully the Slack messages will give me some idea of how prevalent this issue is, and whether we need to dig further.